### ISSUE TITLE
Squiz.WhiteSpace.OperatorSpacing false positive for negation in arrow function

### ISSUE THREAD

**Issue by mabar** [squizlabs/PHP_CodeSniffer#3043]
**Describe the bug**
Minus `-` used as negation of brackets result is okay in `function` but not in `fn`

**Code sample**

This **is** reported
```php
static fn(DateTime $a, DateTime $b): int => -($a->getTimestamp() <=> $b->getTimestamp())
```

This **is not** reported
```php
static function (DateTime $a, DateTime $b): int {
	return -($a->getTimestamp() <=> $b->getTimestamp());
}
```

**Custom ruleset**
```xml
<ruleset>
  <rule ref="Squiz.WhiteSpace.OperatorSpacing">
		<properties>
			<property name="ignoreNewlines" value="true"/>
		</properties>
	</rule>
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above
2. Create a file called `ruleset.xml` with custom ruleset above
2. Run `phpcs test.php --standard=ruleset.xml`
3. See error message displayed
```
Expected 1 space after "-"; 0 found (Squiz.WhiteSpace.OperatorSpacing.NoSpaceAfter)
```

**Expected behavior**

Code is not reported as invalid or at least have same behavior for both contexts (`function` and `fn`)

**Comment by jrfnl** [comment]
Confirmed. I have a fix ready for this, but the PR would conflict with open PRs #3102 and #3103, so I'll wait with pulling it until those have been merged.

**Comment by jrfnl** [comment]
As PRs #3102 and #3103 have been merged, I've pulled the fix for this issue - PR #3129. Testing appreciated.
